pallets/gear/src/manager.rs

```diff
@@ -315,8 +315,27 @@ where
         }
     }
 
-    fn exit_dispatch(&mut self, id_exited: ProgramId, value_destination: ProgramId) {
+    fn exit_dispatch(
+        &mut self,
+        id_exited: ProgramId,
+        message_id: MessageId,
+        value_destination: ProgramId,
+    ) {
         let program_id = id_exited.into_origin();
+        log::debug!("Exit program {:?}", &program_id);
+
+        let message_id = message_id.into_origin();
+        if let Ok(Some(gas_left)) = T::GasHandler::get_limit(message_id) {
```

Comment:
if we pass `gas_amount.left()` into this function in `processor.rs` we'll not need to run this 

```
T::GasHandler::get_limit(message_id)
```

again